src/mixins/Syncable.ts

```diff
@@ -0,0 +1,140 @@
+  registerSyncHandlers(): void;
+  destroy(): void;
+}
+
+export interface SyncDataFull {
+  time: number;
+  playing: boolean;
+  speed: number;
+}
+
+export type SyncData = Partial<SyncDataFull>;
+
+/**
+ * Sync group of tags with each other; every tag should be registered
+ */
+export class SyncManager {
+  syncTargets = new Map<string, Instance<typeof SyncableMixin>>();
+  locked: string | null = null; // refers to the main tag, which locked this sync
+
+  register(syncTarget: Instance<typeof SyncableMixin>) {
+    this.syncTargets.set(syncTarget.name, syncTarget);
+  }
+
+  unregister(syncTarget: Instance<typeof SyncableMixin>) {
+    this.syncTargets.delete(syncTarget.name);
+    // @todo remove manager on empty set
+  }
+
+  /**
+   * Sync `origin` state (in `data`) to connected tags.
+   * No back-sync to origin of the event.
+   * During SYNC_WINDOW only events from origin are processed, others are skipped
+   * @param {SyncData} data state to sync between connected tags
+   * @param {string} event name of event, supplementary info, actions should rely on data
+   * @param {string} origin name of the tag triggered event
+   */
+  sync(data: SyncData, event: SyncEvent, origin: string) {
+    if (!this.locked || this.locked === origin) console.log('SYNC', { event, locked: this.locked, data, origin });
+
+    // locking mechanism
+    // let's try to also send events came from original tag even when sync window is locked
+    if (this.locked && this.locked !== origin) return;
+    if (!this.locked) setTimeout(() => this.locked = null, SYNC_WINDOW);
+    this.locked = origin;
+
+    for (const target of this.syncTargets.values()) {
+      if (origin !== target.name) {
+        target.syncReceive(data, event);
+      }
+    }
+  }
+}
+
+export const SyncManagerFactory = {
+  managers: new Map<string, SyncManager>(),
+  get(name: string): SyncManager {
+    if (!this.managers.has(name)) {
+      this.managers.set(name, new SyncManager());
+    }
+    return this.managers.get(name)!;
+  },
+};
+
+(global as any).syncManagers = SyncManagerFactory;
+
+export type SyncHandler = (data: SyncData, event: string) => void
+
+interface SyncableProps {
+  syncHandlers: Map<string, SyncHandler>;
+  syncManager: SyncManager | null;
+}
+
+/**
+ * Tag should override `registerSyncHandlers()` or `syncReceive()` to handle sync events.
+ * To trigger sync events internal methods should call `syncSend()`.
+ * Should be used before ObjectBase to not break FF_DEV_3391.
+ */
+const SyncableMixin = types
+  .model('SyncableMixin', {
+    name: types.string,
+    sync: types.optional(types.string, ''),
+  })
+  /* eslint-disable @typescript-eslint/indent */
+  .volatile<SyncableProps>(() => ({
+    syncHandlers: new Map(),
+    syncManager: null,
+  }))
+  /* eslint-enable @typescript-eslint/indent */
+  .actions(self => ({
+    afterCreate() {
+      if (!self.sync) return;
+
+      // @todo support sync by other tags' names for backward compatibility
```

Comment:
I think this has still yet to be done for the existing use cases to work, but even the most common use case is Video Timeline Segmentation from our own templates, that Video <-> Audio sample config doesn't work currently.